// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

option go_package = "ebpf.net/collector";

package collector;

message OwnerInfo {
  string uid = 1;
  string name = 2;
  string kind = 3;
}

message ContainerInfo {
  string id = 1;
  string name = 2;
  string image = 3;
}

message PodInfo {
  string uid = 1;
  string ip = 2;
  string name = 6;  // Used when we don't have OWNER. 
  OwnerInfo owner = 3;
  string ns = 4;  // namespace
  string version = 8;
  bool is_host_network = 5;
  repeated string containers_ids = 7; // deprecated
  repeated ContainerInfo container_infos = 9;
}

message ReplicaSetInfo {
  string uid = 1;

  OwnerInfo owner = 2;
}
